src/app/Http/Controllers/BusinessTripController.php

```diff
@@ -35,46 +51,236 @@ public function create() {
+        }
+
+        // Download the file
+        return response()->download($filePath, $trip->upload_name);
     }
 
     /**
      * Returning the view with the trip editing form
+     * @throws \Exception
      */
-    public function edit(BusinessTrip $b) {
-        return view();
+    public function edit(BusinessTrip $trip) {
+        $startDate = new DateTime($trip->datetime_start);
+        $endDate = new DateTime($trip->datetime_end);
+        $interval = new DateInterval('P1D');
+        $dateRange = new DatePeriod($startDate, $interval, $endDate);
+
+        $days = iterator_count($dateRange);
+
+        // Get the collection of contributions associated with the trip
+        $tripContributions = $trip->contributions()->pluck('contributions.id');
+
+        return view('business-trips.edit',  [
+            'trip' => $trip,
+            'days' => $days,
+            'tripContributions' => $tripContributions,
+        ]);
+
     }
 
     /**
-     * Validation as in create(), should check if user is admin/not
      * Redirecting to the trip editing form
      * Sending mail with mail component to admin
      */
-    public function update(Request $request, BusinessTrip $b) {
+    public function update(Request $request, BusinessTrip $trip) {
+        //Validate data
+        $validatedData = $request->validate([
+            'user_id' => 'required|integer|min:0',
+            'country_id' => 'required|exists:countries,id',
+            'transport_id' => 'required|exists:transports,id',
+            'place' => 'required|string|max:200',
+            'event_url' => 'nullable|string|max:200',
+            'upload_name' => 'nullable|string|max:200',
+            'sofia_id' => 'nullable|string|max:40',
+            'state' => 'required|integer',
+            'datetime_start' => 'required|date',
+            'datetime_end' => 'required|date|after:datetime_start',
+            'place_start' => 'nullable|string|max:200',
+            'place_end' => 'nullable|string|max:200',
+            'datetime_border_crossing_start' => 'nullable|date',
+            'datetime_border_crossing_end' => 'nullable|date',
+            'trip_purpose_id' => 'required|integer|min:0', //possibly |exists:trip_purposes,id
+            'purpose_details' => 'nullable|string|max:50',
+            'iban' => 'required|string|max:34',
+            'conference_fee_id' => 'nullable|integer|min:0',
+            'reimbursement_id' => 'nullable|integer|min:0',
+            'spp_symbol_id' => 'nullable|exists:spp_symbols,id',
+            'accommodation_expense_id' => 'nullable|integer|min:0',
+            'travelling_expense_id' => 'nullable|integer|min:0',
+            'other_expense_id' => 'nullable|integer|min:0',
+            'advance_expense_id' => 'nullable|integer|min:0',
+            'not_reimbursed_meals' => 'nullable|string|max:255',
+            'meals_reimbursement' => 'boolean',
+            'expense_estimation' => 'nullable|string|max:20',
```

Comment:
taktiez toto nie je jednoducha hodnota vo formulari. su tam polia {expense_name}_expense_eur {expense_name}_expense_foreign a {expense_name}_reimburse, kde expense_name su kluce v tomto poli: $expenses = ['travelling' => 'Cestovné', 'accommodation' => 'Ubytovanie', 'allowance' => 'Záloha za cestu', 'advance' => 'Vložné', 'other' => 'Iné'];.